src/backend/catalog/aclchk.c

```diff
@@ -5005,14 +5006,31 @@ RemoveRoleFromInitPriv(Oid roleid, Oid classid, Oid objid, int32 objsubid)
 	 * Generate new ACL.  Grantor of rights is always the same as the owner.
 	 */
 	if (old_acl != NULL)
+	{
+		Oid sysadminOid;
+
+		/*
+		 * For babelfish database, grantor will be sysadmin instead of object owner.
+		 */
+		if (bbf_get_sysadmin_oid_hook &&
+			classid == DatabaseRelationId &&
+			objid == get_database_oid(GetConfigOption("babelfishpg_tsql.database_name", true, false), true) &&
```

Comment:
What if `GetConfigOption("babelfishpg_tsql.database_name", true, false)` returns NULL?
Can you please include some manual testing from PG endpoint in the description which makes sure that it is not interfering when hitting this code path from different physical database?